src/statefulmodelinstance.cpp

```diff
@@ -107,13 +107,16 @@ Status StatefulModelInstance::infer(const tensorflow::serving::PredictRequest* r
     if (!status.ok())
         return status;
 
-    MutexPtr sequenceMutexPtr = nullptr;
     std::unique_lock<std::mutex> sequenceManagerLock(sequenceManager->getMutex());
-    status = sequenceManager->getSequenceMutexPtr(sequenceProcessingSpec, sequenceMutexPtr);
+    status = sequenceManager->processRequestedSpec(sequenceProcessingSpec);
     if (!status.ok())
         return status;
+    const uint64_t& sequenceId = sequenceProcessingSpec.getSequenceId();
+    if (!sequenceManager->sequenceExists(sequenceId))
```

Comment:
We have 2 methods that do the same - `sequenceExists` and `hasSequence`. In HasSequence you return INTERNAL_ERROR if sequence is missing, why not to use it here? And get rid of one of the methods.

Reply:
sequenceExists is the low level method to just check if element exists in the sequences container (returning true or false).

hasSequence has more context and returns appropriate status depending on sequence presence in sequences container as well as its termination status (if the last request of the sequence is being processed, the sequence is not available for any other requests).